File: android2_lesson03/src/java/java/com/example/ama/android2_lesson03/ui/search/base/Controller.java
package com.example.ama.android2_lesson03.ui.search.base;

import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.model.LatLng;

public interface Controller {
    void attachMap(GoogleMap map);

    void setLocationAccess(boolean enabled);

    void onResume();

    void onPause();

    void saveState();

    void showOnInnerMap(String markerTitle, String address, LatLng latLng);

    void moveMapCamera(LatLng latLng, float zoom, boolean cameraAnimation);

    void prepareToGMaps();

    interface ClearAddressCallback {
        void clearAddress();
    }
}

File: android2_lesson03/src/kotlin/java/com/example/ama/android2_lesson03/ui/search/base/Controller.kt
package com.example.ama.android2_lesson03.ui.search.base

import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.model.LatLng

interface Controller {
    fun attachMap(map: GoogleMap)
    fun saveState()
    fun onResume()
    fun onPause()
    fun setLocationAccess(enabled: Boolean)
    fun showOnInnerMap(markerTitle: String?, address: String, latLng: LatLng)
    fun moveMapCamera(latLng: LatLng, zoom: Float, cameraAnimation: Boolean)
    fun prepareToGMaps()
}